install/autodetect.go

```diff
@@ -122,24 +129,20 @@ func (k *K8sInstaller) autodetectAndValidate(ctx context.Context) error {
 		}
 	}
 
-	if k.params.IPAM == "" {
-		switch k.flavor.Kind {
-		case k8s.KindKind:
-			k.params.IPAM = ipamKubernetes
-		case k8s.KindEKS:
-			k.params.IPAM = ipamENI
-		case k8s.KindGKE:
-			k.params.IPAM = ipamKubernetes
-		case k8s.KindAKS:
-			k.params.IPAM = ipamAzure
-		default:
-			k.params.IPAM = ipamClusterPool
+	// When on AKS, we additionally need to determine if the cluster is in BYOCNI
+	// mode before determining which IPAM and tunnel type to use.
+	if k.flavor.Kind == k8s.KindAKS {
+		if err := k.azureAutodetect(ctx); err != nil {
+			return err
 		}
-
-		k.Log("🔮 Auto-detected IPAM mode: %s", k.params.IPAM)
 	}
 
 	k.detectDatapathMode(true)
+	// TODO: remove when removing "ipam" flag (marked as deprecated), kept for
+	// backwards compatibility
+	if k.params.IPAM != "" {
+		k.Log("🔮 Custom IPAM mode: %s", k.params.IPAM)
```

Comment:
I think that the "🔮" emoji was intended to communicate that the tool is autodetecting/inferring something from the environment, should this be something like "✅" to acknowledge the user's configuration?